### PR TITLE
Adds visibility check for fully qualified function

### LINKED ISSUE #4231: Visibility qualifier ignored when fully qualified paths are used
For example, say we have a library `pkga` with a private function `foo`:
```rust
library pkga;

fn foo() { }
```
The following script fails to compile because `foo` is not `pub`:
```rust
script;

dep pkga;
use pkga::foo;

fn main() {
    foo();
}
```
However, this compiles fine:
```rust
script;

dep pkga;

fn main() {
    pkga::foo();
}
```

We should also check if this is a problem for other items that accept a visibility qualifier such as structs and enums.

### PR DESCRIPTION
**Pull request by esdrubal**
## Description
The issue was that it was possible to call private methods in another module by calling them with fully qualified paths.

The fix was to use resolve_call_path_with_visibility_check instead of resolve_call_path.

Closes #4231.

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by mohammadfawaz — DISMISSED** [review]
Nice, thanks!

File: test/src/e2e_vm_tests/test_programs/should_fail/private_function_fully_qualified/json_abi_oracle.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (mohammadfawaz) [inline test/src/e2e_vm_tests/test_programs/should_fail/private_function_fully_qualified/json_abi_oracle.json:1]: Actually this file can probably be removed?

Comment (esdrubal) [reply test/src/e2e_vm_tests/test_programs/should_fail/private_function_fully_qualified/json_abi_oracle.json]: Good point, removed file.

**Review by mohammadfawaz — APPROVED** [review]

**Review by nfurfaro — APPROVED** [review]
LGTM.

**Pull request merged**